File: app/src/main/java/com/wangxingxing/easyhttpdemo/http/exception/ResultException.java
package com.wangxingxing.easyhttpdemo.http.exception;

import androidx.annotation.NonNull;

import com.hjq.http.exception.HttpException;
import com.wangxingxing.easyhttpdemo.http.model.HttpData;

/**
 * author : 王星星
 * date : 2023/10/24 17:56
 * email : 1099420259@qq.com
 * description : 返回结果异常
 */
public final class ResultException extends HttpException {

    private final HttpData<?> mData;

    public ResultException(String message, HttpData<?> data) {
        super(message);
        mData = data;
    }

    public ResultException(String message, Throwable cause, HttpData<?> data) {
        super(message, cause);
        mData = data;
    }

    @NonNull
    public HttpData<?> getHttpData() {
        return mData;
    }
}

File: app/src/main/java/com/wangxingxing/easyhttpdemo/http/exception/ResultException.kt
package com.wangxingxing.easyhttpdemo.http.exception

import com.hjq.http.exception.HttpException
import com.wangxingxing.easyhttpdemo.http.model.HttpData

/**
 * author : 王星星
 * date : 2023/10/25 11:44
 * email : 1099420259@qq.com
 * description : 返回结果异常
 */
class ResultException : HttpException {
    private val mData: HttpData<*>

    constructor(message: String?, data: HttpData<*>) : super(message) {
        mData = data
    }

    constructor(message: String?, cause: Throwable?, data: HttpData<*>) : super(message, cause) {
        mData = data
    }

    fun getHttpData(): HttpData<*> {
        return mData
    }
}